### PR TITLE
fix: Requesting ERC-20 or ERC-677 Transfer events in log filter.

### PR DESCRIPTION
**Pull request by mariuspod**
This PR fixes some issues in the treasury_exporter for ERC-677 transfer events which have a different signature than usual Transfer events:
`function transferAndCall(address receiver, uint amount, bytes data) returns (bool success)`

https://github.com/ethereum/EIPs/issues/677

Prior to this fix, these events fail the exporter because the topic filter for the usual ERC-20 transfer event doesn't match with the first topic of the received event.

### CODE REVIEW

File: yearn/treasury/treasury.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 from yearn.exceptions import PriceError
 from yearn.utils import contract
 
-from ..constants import TREASURY_WALLETS
+from ..constants import TREASURY_WALLETS, ERC20_TRANSFER_EVENT_HASH, ERC677_TRANSFER_EVENT_HASH
```
Comment (banteg) [inline yearn/treasury/treasury.py:19]: avoid relative imports if possible

**Comment by banteg** [comment]
I'm looking at the specific example with LINK and it appears we just have the wrong ABI cached.

```
>>> Contract('0x514910771AF9Ca656af840dff83E8264EcF986CA')
<ERC20 Contract '0x514910771AF9Ca656af840dff83E8264EcF986CA'>  # wrong

>>> Contract('0x514910771AF9Ca656af840dff83E8264EcF986CA')
<LinkToken Contract '0x514910771AF9Ca656af840dff83E8264EcF986CA'>  # right
```

File: yearn/treasury/treasury.py [hunk 2]
```diff
@@ -86,15 +89,26 @@ class Treasury:
...
-        ]  # Transfers out of Yearn wallets
+        self._start_block = 10502337 # Treasury didn't exist prior to block 10502337
+
+        # define transfer signatures for Transfer events from ERC-20 and ERC-677 contracts
+        transfer_sigs = [
+            ERC20_TRANSFER_EVENT_HASH,
+            ERC677_TRANSFER_EVENT_HASH
+        ]
+        treasury_addresses = ['0x000000000000000000000000' + address[2:] for address in self.addresses]
+        self._topics = [
```
Comment (banteg) [inline yearn/treasury/treasury.py:100]: this part is great, didn't know it was possible

File: yearn/treasury/treasury.py [hunk 3]
```diff
@@ -62,21 +62,24 @@ def _get_price(token, block=None):
 
 def get_token_from_event(event):
     try:
-        address = event['Transfer'][0].address
+        if '(unknown)' in event:
```
Comment (banteg) [inline yearn/treasury/treasury.py:65]: this part is unneeded, the unknown error comes from wrong LINK abi cached, need to find the root cause of it

**Review by banteg — CHANGES_REQUESTED** [review]

File: yearn/treasury/treasury.py (hunk 3 above)
Comment (mariuspod) [reply yearn/treasury/treasury.py]: removed it, once it pops up again we know what to do.

File: yearn/treasury/treasury.py (hunk 1 above)
Comment (mariuspod) [reply yearn/treasury/treasury.py]: changed

File: yearn/treasury/treasury.py (hunk 2 above)
Comment (mariuspod) [reply yearn/treasury/treasury.py:94]: using encode_* helper methods like you proposed

**Comment by mariuspod** [comment]
@banteg 
updated according to recommendations from review

**Review by banteg — APPROVED** [review]

**Pull request merged**